### PR TITLE
Additional language on dialpad

### PR DESCRIPTION
**Pull request by Belyavin**
Fixes FossifyOrg/Phone#47

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -24,6 +24,8 @@
     <string name="call_number">Call number</string>
     <string name="dialpad_beeps">Enable beeps on dialpad button clicks</string>
     <string name="dialpad_vibrations">Enable vibrations on dialpad button clicks</string>
+    <string name="secondary_dialpad_language">Secondary dialpad language</string>
+    <string name="none_explicit">none</string>
```
Comment (Aga-C) [inline app/src/main/res/values/strings.xml:28]: Instead of creating a new string, please use `none` from Commons.

**Comment by Aga-C** [comment]
Letters in both languages should be centered (e.g. 3 looks really bad).

image

**Comment by derrohrbach** [comment]
@Aga-C also the height of the lowest row of keys is different, which looks weird. 

And also # and * are not vertically aligned with the 0, but this is also the case on the master branch. Might be a different PR?

**Comment by Belyavin** [comment]
I've changed the string to a common one, and aligned the text to the center. I also aligned * and # to 0, though I'm not sure if it became better, need you opinion on this.

**Comment by Belyavin** [comment]
New text alignment

**Review by HenriDellal — CHANGES_REQUESTED** [review]

**Comment by Belyavin** [comment]
Hi, are we waiting for something before merging?

**Comment by Aga-C** [comment]
@Belyavin You have to wait for the final review from Naveen.

File: app/src/main/res/layout/activity_settings.xml [hunk 2]
```diff
@@ -244,8 +255,31 @@
...
+                    android:text="@string/secondary_dialpad_language" />
+
+                <org.fossify.commons.views.MyTextView
+                    android:id="@+id/settings_dialpad_secondary_language"
+                    style="@style/SettingsTextValueStyle"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_below="@+id/settings_dialpad_secondary_language_label"
+                    tools:text="@string/none_explicit" />
+
```
Comment (naveensingh) [inline app/src/main/res/layout/activity_settings.xml:278]: This is still referencing the deleted `none_explicit` string.

File: app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                language.letters[i] = letters
+                for (c in letters) {
+                   language.charMap[c] = i + 2
+                }
+                i++
+            }
+            languages[lang] = language
+        }
+
+        Initialized = true;
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt:65]: Useless `;`

File: app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    public fun convertLettersToNumbers(letters: String, language: String): String {
+        var res = ""
+        val lang = languages[language]
+        val langEn = languages[LOCALE_EN]
+
+        for (c in letters) {
+            if (lang != null && lang.charMap.containsKey(c)) {
+                res += lang.charMap[c]
+            }
+            else if (langEn != null && langEn.charMap.containsKey(c)) {
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt:86]: The code in `DialpadT9.kt` isn't properly formatted in many places. 

https://stackoverflow.com/questions/16580171/code-formatting-shortcuts-in-android-studio-for-operation-systems

File: app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    private val languages = HashMap<String, T9Language>()
+
+    /**
+     * Create a map for Latin (English) characters according to ITU E.161 and ISO/IEC 9995-8
+     */
+    private fun initLatinChars() {
+        val lang = T9Language()
+        for (c in 'A'..'Z') {
+            val number = PhoneNumberUtils.convertKeypadLettersToDigits(c.toString()).toInt();
+            if(number in 2..9) {
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt:27]: There is no space between `if` and `(`...

File: app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    public var Initialized = false
+
+    init {
+        initLatinChars()
+    }
+
+    /**
+     * Read extra languages data from JSON
+     */
+    public fun readFromJson(jsonText: String){
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/phone/helpers/DialpadT9.kt:45]: There is no space between `)` and `{`...

File: app/src/main/res/layout/dialpad.xml [hunk 7]
```diff
@@ -60,9 +60,9 @@
             android:layout_below="@+id/dialpad_2"
```
Comment (naveensingh) [inline app/src/main/res/layout/dialpad.xml:60]: The attribute `android:layout_below` is not valid in the `LinearLayout`. It's there in every button from `dialpad_2_holder` to `dialpad_0_holder` except the asterisk.

File: app/src/main/res/layout/dialpad.xml [hunk 8]
```diff
@@ -271,9 +271,9 @@
             android:layout_height="wrap_content"
             android:layout_centerInParent="true"
```
Comment (naveensingh) [inline app/src/main/res/layout/dialpad.xml:272]: `android:layout_centerInParent="true"` is not valid in the LinearLayout.

File: app/src/main/res/layout/dialpad.xml [hunk 9]
```diff
@@ -322,9 +322,9 @@
...
```
Comment (naveensingh) [inline app/src/main/res/layout/dialpad.xml:322]: `android:layout_centerInParent="true"` is not valid in the LinearLayout.

File: app/src/main/res/layout/dialpad.xml [hunk 10]
```diff
@@ -343,5 +343,5 @@
...
```
Comment (naveensingh) [inline app/src/main/res/layout/dialpad.xml:343]: `android:layout_centerInParent="true"` is not valid in the LinearLayout.

**Review by naveensingh — CHANGES_REQUESTED** [review]

**Comment by naveensingh** [comment]
> I also aligned * and # to 0, though I'm not sure if it became better, need you opinion on this.

@Belyavin indeed * and # look better when centered, I reverted that change.

**Pull request closed without merging**